### PR TITLE
Reporters trigger 1

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: ureport/contacts/migrations/0004_auto_20150918_0915.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+TRIGGER_SQL = """
+-----------------------------------------------------------------------------
+-- Increment or decrement the reporters count on counter of given type
+-----------------------------------------------------------------------------
+CREATE OR REPLACE FUNCTION
+  ureport_insert_reporters_counter(_org_id INT, _type VARCHAR, _count INT)
+RETURNS VOID AS $$
+BEGIN
+  INSERT INTO contacts_reporterscounter("org_id", "type", "count") VALUES(_org_id, _type, _count);
+  PERFORM ureport_may_be_squash_reporterscounters(_org_id, _type);
```
Comment (nicpottier) [inline ureport/contacts/migrations/0004_auto_20150918_0915.py]: ```maybe``` is one word.

Comment (norkans7) [inline ureport/contacts/migrations/0004_auto_20150918_0915.py]: ok

File: ureport/contacts/migrations/0004_auto_20150918_0915.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+      RETURNING "count")
+      INSERT INTO contacts_reporterscounter("org_id", "type", "count")
+      VALUES (_org_id, _type ,GREATEST(0, (SELECT SUM("count") FROM deleted)));
+  END IF;
+END;
+$$ LANGUAGE plpgsql;
+-----------------------------------------------------------------------------
+-- Increment counters for a contact
+-----------------------------------------------------------------------------
+CREATE OR REPLACE FUNCTION ureport_increment_counter_for_contact(_contact contacts_contact, _add BOOLEAN)
```
Comment (nicpottier) [inline ureport/contacts/migrations/0004_auto_20150918_0915.py]: This approach seems like a lot more updates than strictly necessary. You are decrementing and incrementing all possible field values for any contact change?

Couldn't you do this based on whether the field changed in an update/insert?

**Comment by nicpottier** [comment]
This looks really good though I think you should try tweaking the triggers so we don't have to do full decrement and increment of counts on every contact field, instead just doing those that have changed.

File: ureport/contacts/migrations/0004_auto_20150918_0915.py (hunk 2 above)
Comment (norkans7) [inline ureport/contacts/migrations/0004_auto_20150918_0915.py]: Ok, I'll add another method to use on UPDATE that considers the values on OLD and NEW

**Comment by nicpottier** [comment]
Go for it.

**Pull request merged**